docs/development/extensions-core/kubernetes.md

```diff
@@ -57,3 +55,34 @@ Additionally, this extension has following configuration.
 |`druid.discovery.k8s.renewDeadline`|`Duration`|Lease renewal period used by Leader.|PT17S|No|
 |`druid.discovery.k8s.retryPeriod`|`Duration`|Retry wait used by Leader Election algorithm on failed operations.|PT5S|No|
 
+### Gotchas
+
+- Label/Annotation path in each pod spec MUST EXIST, which is easily satisfied if there is at least one label/annotation in the pod spec already. This limitation may be removed in future.
+- Druid Pods need permissions to be able to add labels to self-pod, List and Watch other Pods, create ConfigMap for leader election. Assuming, "default" service account is used by Druid pods, you might need to add following or something similar Kubernetes Role and Role Binding.
```

Comment:
Just curious. Which kind of druid server needs these `add labels to self-pod, List and Watch other Pods, create ConfigMap` permissions? MiddleManager or all the druid servers.

Reply:
everyone needs pod list/update .
`create/update ConfigMap` is needed only by coordinators/overlords which participate in leader election ... but `read ConfigMap` is needed by all so as to know who the leader is. That said, I haven't tested such granularity and would just give same permission to all pods.